### DISCUSSION TITLE
Pass parameters in the web server

### DISCUSSION THREAD

**Discussion by MattiaCC93** [khoih-prog/ESP_WiFiManager#96] [Ideas]
Hello,
I have this need: in my **core.ino** I have defined a **custom function that returns a string** and I would like this string to be displayed by the web server, when it is in AP mode.

I understood how the html page is built (I'm familiar with the ESP_WiFiManager-Impl.h and ESP_WiFiManager.hpp files), what i can't do is to pass this string (computed by my function in core.ino) in the html code.

![image](https://user-images.githubusercontent.com/38031049/198836221-ae79a0a6-756f-42cb-8755-f1bab95f4f48.png)

For the curious, my function returning the string is this:

`String getNumericUID(){  
  byte mac[6];  
  WiFi.macAddress(mac);  
  String ret = "";
  ret =  String(mac[5]) + String(mac[4])+ String(mac[3]) + String(mac[2])+ String(mac[1]) + String(mac[0]);
  return ret;
}`

Note: I am editing your libraries using github's "submodules", they are very handy!

Thank you

**Comment by khoih-prog** [comment]
Sorry I have no time here now, you have to investigate and test yourself. Anything is possible, with correct lines of your code.

Good Luck,

**Comment by MattiaCC93** [comment]
Thank you. I solved it by simply defining the function  within the file ESP_WiFiManager.hpp.
